external-crates/move/crates/move-compiler/src/expansion/aliases.rs

```diff
@@ -43,27 +43,51 @@ impl AliasSet {
     }
 }
 
-macro_rules! resolve_alias {
-    ($map:expr, .$namespace:ident, $name:expr) => {{
-        let mut current_scope = Some($map);
-        let name = $name;
-        loop {
-            let Some(scope) = current_scope else {
-                break None;
-            };
-            let Some(entry) = scope.$namespace.get(name).copied() else {
-                current_scope = scope.previous.as_mut().map(|x| &mut **x);
-                continue;
-            };
-            // Note, might have already been removed by a different `$namespace` resolution
-            scope.unused.remove(&(*name, entry).into());
-
-            let original_name = scope.$namespace.get_full_key(&name).unwrap();
-            break Some((original_name, entry));
+macro_rules! decl_resolve_alias {
```

Comment:
Oof this is a pretty gnarly macro. 

I appreciate that this approach saves a little boilerplate code, but I think it will be easier to maintain if give it a macro flag, where the main call is `find_scope` which takes the enum variant and hands back the scope containing the thing, if any. It will take a bit of massaging, though.

Reply:
I'll see what I can do